docs/coreneuron/inputs.rst

```diff
@@ -0,0 +1,171 @@
+CoreNeuron Inputs
+================
+
+Premise
+-------
+
+This document provides a best-effort overview of the inputs used by CoreNeuron. 
+I do not claim complete knowledge of all internal workings; sections marked 
+with ``TODO`` indicate areas where further clarification is needed.
+
+Input Categories
+----------------
+
+CoreNeuron inputs can be divided into two main categories:
+
+1. **Additional configuration values/files**:
+
+   - **sim.conf**: A general configuration file. All values defined here 
+     can also be passed as command-line parameters.
+
+   - **report.conf**: A configuration file specifying the reports that 
+     CoreNeuron needs to emit. This file is used only if Direct Mode is 
+     not enabled. The output routines rely on ``libsonatareport`` and 
+     comply with the Sonata standard: 
+     `Sonata Documentation <https://sonata-extension.readthedocs.io/en/latest/>`_.
+
+2. **Data passed from NEURON to CoreNeuron**:
+
+   NEURON passes input data to CoreNeuron via files located in the 
+   ``datpath`` specified in ``sim.conf`` (or passed as a CLI argument). 
+   These files include:
+
+   - **gid_1.dat, gid_2.dat, gid_3.dat**: Data for each GID passed in three 
+     phases.
+   - **bbcore_mech.dat**: Information about mechanisms.
+   - **files.dat**: TODO — description needed.
+   - **globals.dat**: TODO — description needed.
+
+sim.conf
+--------
+
+The ``sim.conf`` file is a general configuration file used by CoreNeuron. 
+It can be provided either via the CLI option ``--read-config`` or by setting 
+the ``sim_config`` attribute in Python:
+
+.. code-block:: python
+
+    from neuron import coreneuron
+    coreneuron.sim_config = "<path to sim.conf>"
+
+Example ``sim.conf``:
+
+.. code-block:: ini
+
+    outpath='<path where simulation outputs (i.e., reports) are written>'
+    datpath='<path where NEURON writes input data files for CoreNeuron (usually build/coreneuron_input)>'
+    tstop=100.0
+    dt=0.025
+    prcellgid=-1
+    celsius=34.0
+    voltage=-65.0
+    cell-permute=0
+    mpi=1
+    seed=767740
+    report-conf='<path to report.conf file>'
+
+**Parameters explained**:
+
+- ``outpath``: Path where simulation outputs (i.e., reports) are written.
+- ``datpath``: Path where NEURON writes input data files for CoreNeuron.
+- ``tstop``: Total simulation time in milliseconds.
+- ``dt``: Integration time step in milliseconds.
+- ``prcellgid``: If >=0, print state of this GID during simulation.
+- ``celsius``: Temperature of the simulation in °C.
+- ``voltage``: Initial membrane voltage in mV.
+- ``cell-permute``: Index controlling the permutation of cells:
+  - 0 — No permutation
+  - 1 — Optimize node adjacency (CPU, GPU)
+  - 2 — Optimize parent adjacency (GPU)
```

Comment:
Rendering again 😅 For reference, the docs are visible at https://nrn--3641.org.readthedocs.build/en/3641/coreneuron/inputs.html